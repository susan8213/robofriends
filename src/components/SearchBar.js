import React from "react";

const SearchBar = ({ handleSearchChange }) => {
    return (
        <div className="pa2">
            <input 
                className="pa3 ba b--lightest-blue bg-white" 
                type="search"
                placeholder="Search robots"
                onChange={handleSearchChange}
            ></input>
        </div>
    );
}

export default SearchBar;