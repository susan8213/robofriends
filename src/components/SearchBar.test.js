import { shallow } from 'enzyme';
import React from 'react';
import SearchBar from './SearchBar';


describe('<SearchBar />', () =>{

    it('render without crashing', () => {
        expect(shallow(<SearchBar/>).html()).toMatchSnapshot();
    })

    it('should trigger onChange fn while typing in', () => {
        const searchString = "search";
        const mockOnChange = jest.fn();
        const wrapper = shallow(<SearchBar handleSearchChange={mockOnChange} />);
        wrapper.find('input').at(0).simulate('change', {target: {value: searchString}});
        
        expect(mockOnChange).toHaveBeenCalledTimes(1);
    })
})