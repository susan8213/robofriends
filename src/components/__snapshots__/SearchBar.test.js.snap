// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> render without crashing 1`] = `"<div class=\\"pa2\\"><input type=\\"search\\" class=\\"pa3 ba b--lightest-blue bg-white\\" placeholder=\\"Search robots\\"/></div>"`;
